go/felix/felix.go

```diff
@@ -170,49 +171,73 @@ configRetry:
 	buildInfoLogCxt.WithField("config", configParams).Info(
 		"Successfully loaded configuration.")
 
-	// Create a pair of pipes, one for sending messages to the dataplane
-	// driver, the other for receiving.
-	toDriverR, toDriverW, err := os.Pipe()
-	if err != nil {
-		log.WithError(err).Fatal("Failed to open pipe for dataplane driver")
-	}
-	fromDriverR, fromDriverW, err := os.Pipe()
-	if err != nil {
-		log.WithError(err).Fatal("Failed to open pipe for dataplane driver")
-	}
-
-	cmd := exec.Command(configParams.DataplaneDriver)
-	driverOut, err := cmd.StdoutPipe()
-	if err != nil {
-		log.WithError(err).Fatal("Failed to create pipe for dataplane driver")
-	}
-	driverErr, err := cmd.StderrPipe()
-	if err != nil {
-		log.WithError(err).Fatal("Failed to create pipe for dataplane driver")
-	}
-	go io.Copy(os.Stdout, driverOut)
-	go io.Copy(os.Stderr, driverErr)
-	cmd.ExtraFiles = []*os.File{toDriverR, fromDriverW}
-	if err := cmd.Start(); err != nil {
-		log.WithError(err).Fatal("Failed to start dataplane driver")
-	}
-
-	// Now the sub-process is running, close our copy of the file handles
-	// for the child's end of the pipes.
-	if err := toDriverR.Close(); err != nil {
-		cmd.Process.Kill()
-		log.WithError(err).Fatal("Failed to close parent's copy of pipe")
-	}
-	if err := fromDriverW.Close(); err != nil {
-		cmd.Process.Kill()
-		log.WithError(err).Fatal("Failed to close parent's copy of pipe")
+	// Start up the dataplane driver.  This may be the internal go-based driver or an external
+	// one.
+	var dpDriver dataplaneDriver
+	var dpDriverCmd *exec.Cmd
+	if configParams.UseInternalDataplaneDriver {
+		log.Info("Using internal dataplane driver.")
+		markAccept := configParams.NthIPTablesMark(0)
+		markNextTier := configParams.NthIPTablesMark(1)
+		markWorkload := configParams.NthIPTablesMark(2)
```

Comment:
These would be better with some kind of next mark variable that's incremented each time.